src/main/java/ch/wisv/areafiftylan/web/tournament/controller/TournamentController.java

```diff
@@ -42,6 +43,19 @@ public TournamentController(TournamentService tournamentService) {
         return createResponseEntity(HttpStatus.CREATED, "Tournament added.", tournament);
     }
 
+    @PutMapping("/{id}")
+    @PreAuthorize("hasRole('COMMITTEE')")
+    ResponseEntity<?> editTournament(@RequestBody Tournament tournament) {
+        final Tournament finalTournament = tournament;
+        tournamentService.getTournaments()
+                .stream()
+                .filter(dbTournament -> dbTournament.getId().equals(finalTournament.getId()))
+                .findFirst()
+                .orElseThrow(TournamentNotFoundException::new);
+        tournament = tournamentService.addTournament(tournament);
```

Comment:
Please move all the logic to the tournamentService with a method `replaceTournament(id, tournament)` or something